sovrin_client/agent/agent.py

```diff
@@ -246,22 +253,21 @@ def createAndRunAgent(agentClass, name, wallet=None, basedirpath=None,
                       port=None, looper=None, clientClass=Client, bootstrap=True):
     loop = looper.loop if looper else None
     agent = createAgent(agentClass, name, wallet, basedirpath, port, loop,
-                        clientClass)
+                    clientClass)
     runAgent(agent, looper, bootstrap)
     return agent
 
 
-def isSchemaFound(schema, ranViaScript):
-    if not schema:
-        msg = "Schema not found, check if Sovrin is running and " \
-              "agent's identifier is added"
-        msgHalfLength = int(len(msg)/2)
-        errorLine = "-" * msgHalfLength + "ERROR" + "-" * msgHalfLength
-        print("\n" + errorLine)
-        print("Schema not found, check if Sovrin is running and "
-              "agent's identifier is added")
-        print(errorLine + "\n")
-        if ranViaScript:
+async def runBootstap(isMain, bootstrapFunc):
+    isSuccessful = None
+    error = "Agent bootstrap failed: check if Sovrin is running and " \
+            "agent's identifier is added"
+    try:
+        isSuccessful = await bootstrapFunc()
+    except TimeoutError as e:
+        error += " [cause : {}]".format(str(e))
+
+    if not isSuccessful:
+        logger.error(getFormattedErrorMsg(error))
+        if isMain:
```

Comment:
Underlying question is if it's so tied to an agent's bootstrap, and it feels like "infrastructure," shouldn't it be in Agent?